### ISSUE TITLE
Questions regarding pre-trained models iql_2p[1-12].pthw

### ISSUE THREAD

**Issue by akileshbadrinaaraayanan** [facebookresearch/hanabi_SAD#18]
Hi @hengyuan-hu ,

I have a question regarding the difference between the agents `iql` 1-12 pretrained agents, made available as part of this repo. From, looking at the `selfplay.py` `r2d2.py` and the default `iql_2player.sh`; I understand that num linear layers before LSTM is 1, the num of LSTM layers is 2, and the `rnn_hid_dim` is 512. So, is the variation between these agents only in their seeds? If not, what actually varies. https://github.com/facebookresearch/hanabi_SAD/blob/54a8d34f6ab192898121f8d3935339e63f1f4b35/pyhanabi/r2d2.py#L25

**Comment by hengyuan-hu** [comment]
Yes, the only difference here are seeds as mentioned in our paper. The seed affects the initial weights of the neural network and the seeds of the game sequence during training.

**Comment by hengyuan-hu** [comment]
For the models used in Other-Play paper, they have different seeds, depth and architectures. Those models have not been included in this repo.

**Comment by akileshbadrinaaraayanan** [comment]
Thanks, @hengyuan-hu  for answering this query. I am looking at Figure 4 of the OP paper in which there's a description of 1 layer FC and 2 layer FC with/without residual connection. Can you clarify where's this residual connection done?

**Comment by hengyuan-hu** [comment]
the residual connection by-passes the lstm. 
i.e. add
```
if self.skip_connect:
    o = x + o
```
before this line https://github.com/facebookresearch/hanabi_SAD/blob/54a8d34f6ab192898121f8d3935339e63f1f4b35/pyhanabi/r2d2.py#L54

**Comment by akileshbadrinaaraayanan** [comment]
Thanks a lot @hengyuan-hu! Just to confirm the other pre-trained models,
* `aux_2p_[1-13]` is SAD + AUX (from our earlier discussion, you mentioned AUX was trained with SAD by default).
* `sad_2p_[1-13]` is only SAD.
* `vdn_2p_[1-13]` is VDN (i.e without AUX and SAD).

Could you please confirm if this is correct or if those are something else?

**Comment by hengyuan-hu** [comment]
Yes you are right.

**Comment by akileshbadrinaaraayanan** [comment]
Thanks a lot @hengyuan-hu! Closing this now.
